physcraper/__init__.py

```diff
@@ -1406,38 +1345,49 @@ def get_rank_info_from_web(self, taxon_name):
                 {"taxon id": ncbi_id, "lineage": lineage, "rank": lineage2ranks, "taxon name": tax_name}
         return ncbi_id
 
-    def find_tax_id(self, otu_dict_entry=None, acc=None):
-        """ Find the taxon id in the  otu_dict entry or of a Genbank accession number if no name is given.
-        If not already known it will ask ncbi using the accession number
-
-        :param otu_dict_entry: otu_dict entry
-        :param acc: Genbank accession number
-        :return: ncbi taxon id
-        """
-        # debug("find tax id")
-        inputinfo = False
-        if otu_dict_entry is not None or acc is not None:
-            inputinfo = True
-        assert inputinfo is True
-        ncbi_id = None
-        if otu_dict_entry:
-            ncbi_id = otu_dict_entry["^ncbi:taxon"]
-        if ncbi_id is None:
-            gb_id = None
-            if acc:
-                gb_id = acc
-            elif "^ncbi:accession" in otu_dict_entry:
-                gb_id = otu_dict_entry["^ncbi:accession"]
-            if gb_id in self.acc_ncbi_dict:
-                ncbi_id = self.acc_ncbi_dict[gb_id]
-            else:
-                read_handle = self.entrez_efetch(gb_id)
-                ncbi_id = get_ncbi_tax_id(read_handle)
-                self.acc_ncbi_dict[gb_id] = ncbi_id
-            assert ncbi_id is not None
+ #removed function find_tax_id because it wasn't being used
+   
+    def get_tax_seq_acc(self, acc=None):
```

Comment:
This function is basically replacing map_acc_ncbi(), only that I was using different ways of getting the ncbi_id for the web and database version, making use of the ncbi_parser files for the database version.

Reply:
Ah yeah - Think this is the same issue as the repeated information across databases above. When we end up having to do an entrez web search for the taxon id, we get the whole sequence, so I wanted to make sure we were storing that somewhere and not searching again, but I agree that it is basically the same. And that the entrez web search should maybe never happen with local databases anyways?